WordPress/Classes/ViewRelated/Post/Prepublishing/PrepublishingViewController+JetpackSocial.swift

```diff
@@ -29,6 +29,16 @@ extension PrepublishingViewController {
             configureNoConnectionView(for: cell)
         }
     }
+
+    func didTapAutoSharingCell() {
+        guard hasExistingConnections else {
+            return
+        }
+
+        let model = makeAutoSharingModel()
```

Comment:
I didn't want to redo the query, but storing the model as an instance variable feels a bit awkward since I'll have to refresh it as needed. 

Passing the model directly from `configureAutoSharingView` can work as an alternative, by wrapping the entire AutoSharingView in a `Button` and exposing a public `onTap` closure variable so we can immediately use the model within that method, like so:

```swift
func configureAutoSharingView(for cell: UITableViewCell) {
    let viewModel = makeAutoSharingModel()
    let autoSharingView = PrepublishingAutoSharingView(model: viewModel)
    autoSharingView.onTap = { [weak self] in 
        self?.didTapSocialCell(model: model)
    }
    // ...
```

But wrapping the whole View within a `Button` adds this weird highlight behavior, and the tap action won't register if you tap anywhere that's an empty space. So the tap needs to happen on a view component (label, disclosure indicator, etc.).

Anyways, I'll stick with the current solution for now. 😅